utilities/infra.py

```diff
@@ -439,3 +440,34 @@ def create_inference_token(model_service_account: ServiceAccount) -> str:
     return run_command(
         shlex.split(f"oc create token -n {model_service_account.namespace} {model_service_account.name}")
     )[1].strip()
+
+
+def check_pod_status_in_time(pod: Pod, status: Set[str], duration: int = TIMEOUT_2MIN, wait: int = 1) -> None:
+    """
+    Checks if a pod has a given status for a given duration.
```

Comment:
nitpick, but perhaps something like `Checks if a pod reaches a given status withing a given timeout` would be more clear on what this is doing